flocker/node/agents/blockdevice.py

```diff
@@ -230,6 +235,46 @@ def _action(self):
         return DESTROY_BLOCK_DEVICE_DATASET(_logger, volume=self.volume)
 
 
+# Implement:
+#     AttachVolume(volume=volume),
+#     MountVolume(volume=volume),
+# in order to carry out the sequence below.
+# Maybe use those in the CreateBlockDeviceDataset state change or paste in some
+# followup issues, or maybe do those issues first??
+
+
+@_logged_statechange
+@with_cmp(["change"])
+class ResizeBlockDeviceDataset(proxyForInterface(IStateChange, "change")):
```

Comment:
Agreed on this - though as with FLOC-1582 I don't think fixing this pattern needs to block this work.  However, take a look at the changes that were necessary in FLOC-1582.  The shape of `DeleteBlockDeviceDataset` changed a fair bit to account for some other required changes (not doing asynchronous work in `calculcate_necessary_state_changes`).  The same applies here.